src/main/java/org/spongepowered/api/map/font/MapFont.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.map.font;
+
+import com.google.common.collect.ImmutableMap;
+import org.spongepowered.api.CatalogType;
+import org.spongepowered.api.util.annotation.CatalogedBy;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Represents a font on a map that can be used for drawing characters.
+ */
+public final class MapFont {
```

Comment:
It simplifies plugins and let's plugins draw Text in custom ways without
having to parse the whole text datatype. I didn't want a less featured API
than Bukkit had, and this was a useful ability in maps.

I put it in API since its not containing any implementation specific
details and thus could be used by any implementation based around our API.

On Sat, Apr 2, 2016, 3:15 PM Gabriel Harris-Rouquette <
notifications@github.com> wrote:

> In src/main/java/org/spongepowered/api/map/font/MapFont.java
> <https://github.com/SpongePowered/SpongeAPI/pull/1162#discussion_r58299195>
> :
>
> > +package org.spongepowered.api.map.font;
> > +
> > +import com.google.common.collect.
>
> ImmutableMap;
> > +import org.spongepowered.api.CatalogType;
> > +import org.spongepowered.api.util.annotation.CatalogedBy;
> > +
> > +import java.util.Map;
> > +import java.util.Optional;
> > +
> > +import static com.google.common.base.Preconditions.checkArgument;
> > +import static com.google.common.base.Preconditions.checkNotNull;
> > +
> > +/**
> > + * Represents a font on a map that can be used for drawing characters.
> > + */
> > +public final class MapFont {
>
> A lot of this class sounds like implementation details and/or needs to be
> clearly defined that custom MapFonts can be used, but aren't identifiable
> in any way shape or form (i.e. they're not extending CatalogType).
>
> —
> You are receiving this because you authored the thread.
>
> Reply to this email directly or view it on GitHub
> <https://github.com/SpongePowered/SpongeAPI/pull/1162/files/d8338269be828e278d15ed5ffbe9bb30b7c899c1#r58299195>
>